src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -135,20 +140,26 @@ private UnboundLambda AnalyzeAnonymousFunction(
 
                 int parameterCount = 0;
                 int underscoresCount = 0;
-                foreach (var p in parameterSyntaxList.Value)
+                int firstDefault = -1;
+                for (int i = 0, n = parameterSyntaxList.Count; i < n; i++)
                 {
                     parameterCount++;
 
+                    var p = parameterSyntaxList[i];
                     if (p.Identifier.IsUnderscoreToken())
                     {
                         underscoresCount++;
                     }
 
                     checkAttributes(syntax, p.AttributeLists, diagnostics);
 
-                    var isAnonymousMethod = syntax.IsKind(SyntaxKind.AnonymousMethodExpression);
                     if (p.Default != null)
                     {
+                        if (firstDefault == -1)
```

Comment:
this logic moved from teh caller.  no need for it to do a secondary pass to report errors when it can just do that here.